ooni/templates/httpt.py

```diff
@@ -1,23 +1,28 @@
 import re
+import codecs
 import random
 
-from twisted.internet import defer
-
 from txtorcon.interface import StreamListenerMixin
 
+from twisted.web.client import readBody, PartialDownloadError
+from twisted.web.client import ContentDecoderAgent
+
 from twisted.internet import reactor
 from twisted.internet.endpoints import TCP4ClientEndpoint
-from ooni.utils.trueheaders import TrueHeadersAgent, TrueHeadersSOCKS5Agent
+
+from ooni.utils.socks import TrueHeadersSOCKS5Agent
 
 from ooni.nettest import NetTestCase
-from ooni.utils import log, base64Dict
+from ooni.utils import log
 from ooni.settings import config
 
-from ooni.utils.net import BodyReceiver, StringProducer, userAgents
-from ooni.utils.trueheaders import TrueHeaders
+from ooni.utils.net import StringProducer, userAgents
+from ooni.common.txextra import TrueHeaders
+from ooni.common.txextra import FixedRedirectAgent, TrueHeadersAgent
+from ooni.common.http_utils import representBody
 from ooni.errors import handleAllFailures
 
-META_CHARSET_REGEXP = re.compile('<meta(?!\s*(?:name|value)\s*=)[^>]*?charset\s*=[\s"\']*([^\s"\'/>]*)')
+META_CHARSET_REGEXP = re.compile('<meta(?!\s*(?:name|value)\s*=)[^>]*?charset\s*=[\s"\']*([^\s"\'/>!;]+)')
```

Comment:
Yes, it's actually not used in this module at all. It's probably a leftover from when I moved stuff into common. Will fix.